### PR TITLE
fix: DF-148 schools lookup refactor

### PR DESCRIPTION
**Pull request by curlyfriesplease**
#### To pass testing pipeline, these need to follow Conventional Commits spec:
https://www.conventionalcommits.org/en/v1.0.0/
e.g.
`feat: create NewForm component`
or
`fix: update broken link in NewForm component`

#### What is it doing?
There's a fixed black border around the schoolsLookup compo, and no close-on-click-outside behaviour. This fixes both.

#### Why is this required?
The donate FE rebuild doesn't need the border, and needs close on click outside.

#### link to Jira ticket:
[DF-148]

[DF-148]: https://comicrelief.atlassian.net/browse/DF-148?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js [hunk 1]
```diff
@@ -96,36 +121,37 @@ const TextInputWithDropdown = React.forwardRef(
...
   activeOption,
   resetActiveOption,
   ...rest
-}) => {
+}, ref) => {
   // Reset 'activeOption' when the list is unfocused.
   const onBlur = e => {
     const { target } = e;
-    // There's a delay before the new activeOption becomes the document.activeElement when
-    //  scrolling through the dropdown list via keyboard.
```
Comment (AndyEPhipps) [inline src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js:128]: Why remove this?

File: src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js [hunk 2]
```diff
@@ -37,16 +37,41 @@ const TextInputWithDropdown = React.forwardRef(
...
+          && !dropdownRef.current.contains(event.target)
+          && !containerRef.current.contains(event.target)) {
+          setForceClosed(true);
+        }
+      };
+
+      // Only add the listeners if we have options showing
+      if (options.length > 0 && !forceClosed) {
+        document.addEventListener('mousedown', handleClickOutside);
+        document.addEventListener('touchstart', handleClickOutside);
```
Comment (AndyEPhipps) [inline src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js:62]: If we wanted to get real fancy and streamlined

`["mousedown", "touchstart"].forEach(event => document.addEventListener(event, handleClickOutside));`
and
`["mousedown", "touchstart"].forEach(event => document. removeEventListener(event, handleClickOutside));
`

File: src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js [hunk 3]
```diff
@@ -37,16 +37,41 @@ const TextInputWithDropdown = React.forwardRef(
...
+      }
+
+      return () => {
+        document.removeEventListener('mousedown', handleClickOutside);
```
Comment (AndyEPhipps) [inline src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js:66]: 10 points for Gryffindor

**Review by AndyEPhipps — APPROVED** [review]
Makes sense to me, suggested streamlining but NBD really

File: src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js (hunk 1 above)
Comment (curlyfriesplease) [reply src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js:128]: AH dope. I was running through removing comments I'd left myself as I worked, and removed this too in the haste. Apologies

File: src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js (hunk 2 above)
Comment (curlyfriesplease) [reply src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js:62]: Very good suggestion. I like it.

File: src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js (hunk 1 above)
Comment (AndyEPhipps) [reply src/components/Atoms/TextInputWithDropdown/TextInputWithDropdown.js]: always check those mad diffs

**Pull request merged**